### ISSUE TITLE
use tensorflow feature columns

### ISSUE THREAD

**Issue by BenMacKenzie** [ludwig-ai/ludwig#249]
After generating the metadata file, rather then using it to integerize categorical data and replace missing values in the hdf5 file,  you could use it to create appropriate feature columns using the tensorflow estimator api.  This would allow the model to be deployed without needing any data pre-processing at inference time.

**Comment by w4nderlust** [comment]
I'm not a huge fan of the estimator api. Anyway, we are taking a closer look at TensorFlow 2.0, so this may end up being considered. The problem is that the HDF5 solution has  the great advantage that it requires performing preprocessing just one time, so we have to figure out if this advantage can be kept if we use columns.

**Comment by BenMacKenzie** [comment]
looks like keras supports feature columns now...so not really sure what the point of estimator api is.

**Comment by justinxzhao** [comment]
Ludwig has migrated to PyTorch.
